File: src/javascript/syntax.test.js
import functions from './syntax'

test('checking', () => {
   expect(functions.defineNumber(1)).toBe(1);
   expect(functions.defineString("string testing")).toBe("string testing");
   expect(functions.defineBoolean(3,2)).toBe(true);
   expect(functions.defineArray([1,2])).toBe(1);    
   expect(functions.defineObject({name: "Dan", hobby: "Badminton"})).toBe("Dan");   
   expect(functions.defineUndefined("return undefined")).toBe();
   expect(functions.sampleOfIfElse(5,10)).toBe("5 < 10");
   expect(functions.sampleOfIfElse(10,5)).toBe("10 > 5");  
   expect(functions.arrayToFront(["apple"])).toBe("pinnaple, apple");  
   expect(functions.arrayToEnd(["apple"])).toBe("apple, pinnaple");
   expect(functions.arrayUpdateValue(["apple"])).toBe("pear");
   expect(functions.forLoop(3)).toBe(3);
   expect(functions.forInLoop({fname: "Dan", lname: "T"})).toBe("Dan T");
   expect(functions.whileLoop(3)).toBe("The last number is 2");
   expect(functions.doWhileLoop(3)).toBe("The last number is 2");
   expect(functions.forEachWithArrFunction(["a","b","c"])).toBe("abcd");
   expect(functions.declareObject({name: "Dan", gender: "male"})).toBe("Dan is male");
   expect(functions.objectKeyValue({name: "Dan", gender: "male"})).toBe("Dan");

})

File: src/python/syntax_test.py
import syntax

def testNum():
    assert syntax.defineNum(1)==1
    assert syntax.defineString("dd")=="dd"
    assert syntax.defineBoolean(3,1)==True
    assert syntax.defineBoolean(3,5)==False
    assert syntax.defineArray()==['A','B']
    assert syntax.defineDictionary({"name":"Tom", "Age":20})["name"]=="Tom"
    # create an object f the class MyClass
    myObject=syntax.MyClass()
    assert myObject.name=="in Class"
    assert syntax.defineUndefined()=="undefined"
    assert syntax.ifElseSample(5,2)=="5>2"
    assert syntax.ifElseSample(2,5)=="2<5"
    assert syntax.acceptParameters(1,2)=="two parameters are 1 and 2."
    assert syntax.arrAddToFront('A')==['A','B','C']
    assert syntax.arrAddToLast('C')==['A','B','C']
    assert syntax.arrUpdate("A updated to UPDATED")==["A updated to UPDATED","B"]
    assert syntax.testForIn([1,4])==5
    assert syntax.testWhile([1,4,5])==10
    assert syntax.defineObject({"name":"Dan", "hobby":"sport"})=="Dan"